src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs

```diff
@@ -137,14 +137,42 @@ static void Main(string[] args)
     static void Main(string[] args)
     {
         using (null)
-        using (null)$$
+                using (null)$$
     }
 }
 ";
 
             AssertFormatWithPasteOrReturn(expected, code, allowDocumentChanges: true, isPaste: false);
         }
 
+        [WpfFact]
+        [WorkItem(912965, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/912965")]
+        [Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void FormatUsingStatementWhenTypingCloseParen()
+        {
+            var code = @"class Program
+{
+    static void Main(string[] args)
+    {
+        using (null)
+                using (null)$$
+    }
+}
+";
+
+            var expected = @"class Program
+{
+    static void Main(string[] args)
+    {
+        using (null)
+        using (null)
+    }
+}
+";
+
+            AssertFormatAfterTypeChar(code, expected);
```

Comment:
I see so you want to remove some code so changed behavior so that format happen on ")" rather than on enter.

Reply:
if we even see behavior changes on existing tests, I wonder how many behavior changes actually there is that are not covered by tests. I dont believe formatting tests we have would cover like 1% of all possible combination that could happen.